diff --git a/runtime/js/domain.js b/runtime/js/domain.js
index f83d107..beefbfc 100644
--- a/runtime/js/domain.js
+++ b/runtime/js/domain.js
@@ -7,6 +7,7 @@ var caml_domain_dls = [0];
 //Version: >= 5
 function caml_domain_dls_set(a) {
   caml_domain_dls = a;
+  return 0;
 }
 
 //Provides: caml_domain_dls_compare_and_set
@@ -268,3 +269,19 @@ function caml_atomic_lxor_field(ref, field, i) {
   ref[field + 1] ^= i;
   return 0;
 }
+
+//Provides: caml_domain_tls
+var caml_domain_tls = [0];
+
+//Provides: caml_domain_tls_set
+//Requires: caml_domain_tls
+function caml_domain_tls_set(a) {
+  caml_domain_tls = a;
+  return 0;
+}
+
+//Provides: caml_domain_tls_get
+//Requires: caml_domain_tls
+function caml_domain_tls_get(a) {
+  return caml_domain_tls;
+}
