diff --git a/src/ocaml/preprocess/dune b/src/ocaml/preprocess/dune
index 515193c..2e7288a 100644
--- a/src/ocaml/preprocess/dune
+++ b/src/ocaml/preprocess/dune
@@ -8,41 +8,32 @@
 
 (menhir
  (modules parser_raw)
- (enabled_if (<> %{profile} "release"))
- (mode    (promote (only parser_raw.ml parser_raw.mli)))
  (flags :standard --inspection --table --cmly --lalr))
 
 (rule
   (targets parser_recover.ml)
-  (enabled_if (<> %{profile} "release"))
   (deps    parser_raw.cmly)
-  (mode    promote)
   (action
     (with-stdout-to %{targets}
        (run %{exe:./recover/gen_recover.exe} %{deps}))))
 
 (rule
   (targets parser_explain.ml)
-  (enabled_if (<> %{profile} "release"))
   (deps    parser_raw.cmly)
-  (mode    promote)
   (action
     (with-stdout-to %{targets}
        (run %{exe:./explain/gen_explain.exe} %{deps}))))
 
 (rule
   (targets parser_printer.ml)
-  (enabled_if (<> %{profile} "release"))
   (deps    parser_raw.cmly)
-  (mode    promote)
   (action
     (with-stdout-to %{targets}
        (run %{exe:./printer/gen_printer.exe} %{deps}))))
 
 (rule
  (targets menhirLib.ml menhirLib.mli)
- (mode promote)
  (action
   (progn
-   (copy %{lib:menhirLib:menhirLib.ml}  menhirLib.ml)
-   (copy %{lib:menhirLib:menhirLib.mli} menhirLib.mli))))
+   (copy %{workspace_root}/vendor/menhir/lib/pack/menhirLib.ml menhirLib.ml)
+   (copy %{workspace_root}/vendor/menhir/lib/pack/menhirLib.mli menhirLib.mli))))
\ No newline at end of file
diff --git a/src/sherlodoc/dune b/src/sherlodoc/dune
index 6d03533..9f1c194 100644
--- a/src/sherlodoc/dune
+++ b/src/sherlodoc/dune
@@ -4,8 +4,6 @@
 
 (menhir
  (modules type_parser)
- (enabled_if (<> %{profile} "release"))
- (mode promote)
  (flags :standard --explain --lalr))
 
 (ocamllex type_lexer)
